x-pack/metricbeat/magefile.go

```diff
@@ -0,0 +1,270 @@
+}
+
+// GoUnitTest executes the Go unit tests.
+// Use TEST_COVERAGE=true to enable code coverage profiling.
+// Use RACE_DETECTOR=true to enable the race detector.
+func GoUnitTest(ctx context.Context) error {
+	return mage.GoTest(ctx, mage.DefaultGoTestUnitArgs())
+}
+
+// GoIntegTest executes the Go integration tests.
+// Use TEST_COVERAGE=true to enable code coverage profiling.
+// Use RACE_DETECTOR=true to enable the race detector.
+func GoIntegTest(ctx context.Context) error {
+	return mage.RunIntegTest("goIntegTest", func() error {
+		return mage.GoTest(ctx, mage.DefaultGoTestIntegrationArgs())
+	})
+}
+
+// PythonUnitTest executes the python system tests.
+func PythonUnitTest() error {
+	mg.Deps(mage.BuildSystemTestBinary)
+	return mage.PythonNoseTest(mage.DefaultPythonTestUnitArgs())
+}
+
+// PythonUnitTest executes the python system tests in the integration environment (Docker).
+func PythonIntegTest(ctx context.Context) error {
+	if !mage.IsInIntegTestEnv() {
+		mg.Deps(Fields)
+	}
+	return mage.RunIntegTest("pythonIntegTest", func() error {
+		mg.Deps(mage.BuildSystemTestBinary)
+		return mage.PythonNoseTest(mage.DefaultPythonTestIntegrationArgs())
+	})
+}
+
+// -----------------------------------------------------------------------------
+// Customizations specific to Metricbeat.
+// - Include modules.d directory in packages.
+
+const (
+	dirModulesDGenerated = "build/package/modules.d"
+)
+
+// prepareModulePackaging generates modules and modules.d directories
+// for an x-pack distribution, excluding _meta and test files so that they are
+// not included in packages.
+func prepareModulePackaging() error {
+	mg.Deps(createDirModulesD)
+
+	err := mage.Clean([]string{
+		dirModulesDGenerated,
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, copyAction := range []struct {
+		src, dst string
+	}{
+		{mage.OSSBeatDir("modules.d"), dirModulesDGenerated},
+		{"modules.d", dirModulesDGenerated},
+	} {
+		err := (&mage.CopyTask{
+			Source:  copyAction.src,
+			Dest:    copyAction.dst,
+			Mode:    0644,
+			DirMode: 0755,
+		}).Execute()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func shortConfig() error {
+	var configParts = []string{
+		mage.OSSBeatDir("_meta/common.yml"),
+		mage.OSSBeatDir("_meta/setup.yml"),
+		"{{ elastic_beats_dir }}/libbeat/_meta/config.yml",
+	}
+
+	for i, f := range configParts {
+		configParts[i] = mage.MustExpand(f)
+	}
+
+	configFile := mage.BeatName + ".yml"
+	mage.MustFileConcat(configFile, 0640, configParts...)
+	mage.MustFindReplace(configFile, regexp.MustCompile("beatname"), mage.BeatName)
+	mage.MustFindReplace(configFile, regexp.MustCompile("beat-index-prefix"), mage.BeatIndexPrefix)
+	return nil
+}
+
+func referenceConfig() error {
+	const modulesConfigYml = "build/config.modules.yml"
+	err := mage.GenerateModuleReferenceConfig(modulesConfigYml, mage.OSSBeatDir("module"), "module")
+	if err != nil {
+		return err
+	}
+	//defer os.Remove(modulesConfigYml)
```

Comment:
This should be uncommented. We want to remove the file when this function finishes.

Reply:
Ok!